pkg/applicationserver/io/pubsub/provider/mqtt/provider_test.go

```diff
@@ -178,6 +178,9 @@ func TestOpenConnection(t *testing.T) {
 			},
 		},
 	} {
+		client := utc.createClient(t, a)
+		defer client.Disconnect(uint(timeout / time.Millisecond))
```

Comment:
This `defer` is in a `for` loop and will occur at the end of the test-case, but since we don't share the client ID, this is fine.

Reply:
But why not put it in `Run()` below? It's only used in that scope and `defer` would work, right?